.github/workflows/go.yml

```diff
@@ -0,0 +1,26 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Build and Test
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
```

Comment:
we may want to consider adding a `matrix` to test the major OSes in the future, but this looks good for now.